encoding/codecv4/codecv4.go

```diff
@@ -239,6 +248,23 @@ func ConstructBlobPayload(chunks []*encoding.Chunk, enableCompress bool, useMock
 	return blob, blobVersionedHash, &z, blobBytes, nil
 }
 
+// DecodeTxsFromBlob decodes txs from blob bytes and writes to chunks
+func DecodeTxsFromBlob(blob *kzg4844.Blob, chunks []*DAChunkRawTx) error {
+	rawBytes := encoding.BytesFromBlobCanonical(blob)
+
+	// if first byte is 1 - data compressed, 0 - not compressed
+	if rawBytes[0] == 0x1 {
+		magics := []byte{0x28, 0xb5, 0x2f, 0xfd}
+		blobBytes, err := encoding.DecompressScrollBatchBytes(append(magics, rawBytes[1:]...))
```

Comment:
The implementation has no issues. I want to point out the naming here though. We decompress a `blob` to get `batch`. So we should have:
```suggestion
		batchBytes, err := encoding.DecompressScrollBatchBytes(append(magics, rawBytes[1:]...))
```